plugins/catalog-backend-module-terraform-cloud/src/processors/TerraformCloudProcessor.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec, Entity } from '@backstage/catalog-model';
+import {
+  CatalogProcessor,
+  CatalogProcessorEmit,
+  results
+} from '@backstage/plugin-catalog-backend';
+import fetch from 'cross-fetch';
+import { Config } from '@backstage/config';
+import  { Logger } from 'winston';
+
+export class TfCloudReaderProcessor implements CatalogProcessor {
+    constructor(private readonly config: Config, private readonly logger: Logger) {}
+    
+ 
+  async readLocation(
+    location: LocationSpec,
+    _optional: boolean,
+    emit: CatalogProcessorEmit,
+  ): Promise<boolean> {
+    if (location.type !== 'tf-cloud') {
+      return false;
+    }
+        
+    try {
+             const token = this.config.getOptionalString('integrations.terraform.token');
```

Comment:
Not sure about this config pattern tbh, I think either it would just be a top-level `terraform.token`, or use the same layout as the other integrations with an array of different terraform integrations that one can select from. I think the plain `terraform.token` would make sense for now, and then we can see where we end up with more terraform integrations going forward. Another slight spin on that could be `terraform.cloud.token`

Reply:
Makes sense to me - will go with the `terraform.token` as there really isn't any other auth token used in terraform, so `terraform.cloud.token` might be a bit too wordy :)